content selector component

lchan131 opened the pull request.

rcausey commented:
This looks good, but I'm thinking it might be better if we extracted the buttons out of the individual components and implemented them inside of `d2l-content-selector` instead. This would make the individual components more flexible and re-useable outside of a dialog. Would this be a lot of work to change?

lchan131 commented:
> This looks good, but I'm thinking it might be better if we extracted the buttons out of the individual components and implemented them inside of `d2l-content-selector` instead. This would make the individual components more flexible and re-useable outside of a dialog. Would this be a lot of work to change?

No this would be easy to do. I'll work on it

rcausey commented on core/d2l-content-selector/d2l-content-selector.js:
```diff
@@ -1,9 +1,276 @@
+class ContentSelector extends InternalLocalizeMixin(LitElement) {
+	static get properties() {
+		return {
+			allowExistingObjectSelection: { type: Boolean },
```
Same as other comments - I think we should remove this `allowExistingObjectSelection` for now and just default to the behaviour in the SCORM dialog

rcausey commented on core/d2l-content-selector/d2l-content-selector.js:
```diff
@@ -1,9 +1,276 @@
+		this._saveButtonPropertiesDisabled = true;
+		await this.contentProperties.save();
+		this._saveButtonPropertiesDisabled = false;
+		if (this.autoSelectAfterUpload) {
```
Should be removed

rcausey commented on core/d2l-content-selector/d2l-content-selector.js:
```diff
@@ -1,9 +1,276 @@
+	async handleSettingsAddContent() {
+		const content = this.topicSettings.getContent();
+		await this.client.topic.postItem(content);
+		this._openPreview();
```
I don't think openPreview should be done here anymore

lchan131 replied:
The _openPreview function only emits an event, so I think it should stay.

rcausey commented on core/d2l-content-selector/d2l-content-selector.js:
```diff
@@ -1,9 +1,276 @@
+	}
+
+	handleUploadSuccess({ detail: { d2lrn } }) {
+		if (this.editPropertiesAfterUpload) {
```
Default to SCORM dialog behaviour here instead of using these options

rcausey commented on core/d2l-content-topic-settings/d2l-content-topic-settings.js:
```diff
@@ -111,6 +120,9 @@ class ContentTopicSettings extends InternalLocalizeMixin(LitElement) {
 		this.serviceUrl = '';
+		this.contentId = '';
+		this.tenantId = '';
+		this.resourceType = '';
```
See other comments in other component about changing this from resourceType to revisionTag and determining resourceType from the revision

rcausey reviewed (changes requested):
In general, this looks really good. Most of my comments are just because I've changed my mind from the initial spec and want to keep this as simple as possible until we need those other options.

rcausey commented on core/d2l-content-topic-settings/d2l-content-topic-settings.js:
```diff
@@ -92,8 +94,8 @@ class ContentTopicSettings extends InternalLocalizeMixin(LitElement) {
 
-		.action-group {
-			margin-top: 10px;
+		.wrapper {
```
I think this wrapper shouldn't be necessary in here since we moved the buttons outside of the component, right?

rcausey commented on core/d2l-content-topic-settings/d2l-content-topic-settings.js:
```diff
@@ -122,24 +128,35 @@ class ContentTopicSettings extends InternalLocalizeMixin(LitElement) {
+			revision = content.revisions.find(rev => rev.id === this.revisionTag);
+		}
+
+		console.log(revision);
```
Console log

rcausey commented on core/d2l-content-topic-settings/d2l-content-topic-settings.js:
```diff
@@ -122,24 +128,35 @@ class ContentTopicSettings extends InternalLocalizeMixin(LitElement) {
 	render() {
 		return html`
-			<div>
+			<div class="wrapper">
```
See comment above re: wrapper

lchan131 replied:
Again, this is necessary for the demo

rcausey commented on core/d2l-content-properties/demo/index.html:
```diff
@@ -17,7 +17,7 @@
 			<d2l-demo-snippet>
 				<d2l-content-properties
-					d2lrn="d2l:brightspace:content:region:97dc5895-cd3b-465f-8d5b-3c8d7e1bb6a7:scorm:a0605fa8-a348-4361-83b0-925fc57e761e"
+					d2lrn="d2l:brightspace:content:region:97dc5895-cd3b-465f-8d5b-3c8d7e1bb6a7:scorm:0"
```
Doesn't really matter, but just curious why you changed the contentId from a GUID to a number?

lchan131 replied:
I just figured it would be easier in the future if we wanted to add a new type, for instance.

rcausey commented on core/d2l-content-properties/d2l-content-properties.js:
```diff
@@ -98,11 +102,21 @@ class ContentProperties extends InternalLocalizeMixin(LitElement) {
 			font-size: 16px;
 		}
+
+		.settings-container {
```
Let's move this margin out of the component and into the selector instead

lchan131 replied:
Alright. I'll do this for the wrapper classes discussed above as well.

rcausey commented on core/d2l-content-selector/d2l-content-selector.js:
```diff
@@ -1,9 +1,259 @@
+	constructor() {
+		super();
+
+		this.allowExistingObjectSelection = true;
```
Let's get rid of `allowExistingObjectSelection` for now

rcausey commented on core/d2l-content-selector/d2l-content-selector.js:
```diff
@@ -1,9 +1,259 @@
+	connectedCallback() {
+		super.connectedCallback();
+
+		if (this.allowUpload && !this.allowExistingObjectSelection) {
```
See above comment on `allowExistingObjectSelection` - shouldn't need this after removal

rcausey reviewed (approved):
Looks good, just look into my comments before merging
